pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/renderer/TextValueProvider.java

```diff
@@ -8,38 +8,89 @@
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
+import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.broadinstitute.dsm.model.elastic.export.tabular.FilterExportConfig;
 import org.broadinstitute.dsm.statics.ESObjectConstants;
 import org.broadinstitute.dsm.util.ElasticSearchUtil;
 
+/** Base class for rendering participant data as text */
 public class TextValueProvider {
-    public Collection<?> getRawValues(FilterExportConfig filterConfig, Map<String, Object> moduleMap) {
-        Object value = StringUtils.EMPTY;
+
+    /**
+     *
+     * @param filterConfig the config for the question/item
+     * @param formMap a map of the data for this module response.  That should be a value returned from getModuleCompletions
+     *                in TabularParticipantParser
+     * @return a list of lists of response values.  The outer list has one entry for each response given (e.g. for a medication list
+     * question, one entry will correspond to each medication). The outer list will always have only one entry unless
+     * the question is an allowMultiple.  The inner lists corresponds to the values for each response. It will have one entry
+     * for most questions, but it will have multiple entries for either selectMode=Multiple questions or composite questions
+     */
+    public List<List<String>> getFormattedValues(FilterExportConfig filterConfig, Map<String, Object> formMap) {
+        return collectFormattedResponses(getRawValues(filterConfig, formMap), filterConfig, formMap);
+    }
+
+
+    /** looks for additional details associated with an option response.  For example, if the participant was asked to
+     * select a symptom, and then for each symptom selected, enter the age at which the symptom began.
+     */
+    public String getOptionDetails(FilterExportConfig filterConfig, Map<String, Object> moduleMap, String optionStableId, int responseNum) {
+        Map<String, Object> answerObject = getRelevantAnswer(moduleMap, filterConfig);
+        if (answerObject != null) {
+            Object optionDetails = answerObject.get(ESObjectConstants.OPTIONDETAILS);
+            // Option details are stored in ES as an array of maps
+            // e.g. [{option: "TELANGIECTASIA_EYES", details: "12"}, {option: "TELANGIECTASIA_SKIN", details: "34"}]
+            if (optionDetails instanceof List) {
+                return ((List<?>) optionDetails).stream().filter(detail ->
+                                StringUtils.equals((String) ((Map) detail).get(ESObjectConstants.OPTION), optionStableId))
+                        .map(detail -> (String) ((Map<String, Object>) detail).get(ESObjectConstants.DETAILS))
+                        .findAny().orElse(StringUtils.EMPTY);
```

Comment:
I would extract this piece of code in method with name `extractOptionDetails` which is way more better in my opinion than having comment.